### ISSUE TITLE
"Missing content" accessibility audit error on sidebar nav

### ISSUE THREAD

**Issue by NeilJS** [withastro/starlight#3159]
### What version of `starlight` are you using?

0.34.1

### What version of `astro` are you using?

5.7.8

### What package manager are you using?

N/A

### What operating system are you using?

N/A

### What browser are you using?

Chrome

### Describe the Bug

When previewing Starlight via the  "See it in action" [Stackblitz link](https://stackblitz.com/github/withastro/starlight/tree/main/examples/basics), found in the [Getting Started](https://starlight.astro.build/getting-started/) guided (as well as when installed locally) and running the Audit from the toolbar, we immediately see accessibility errors relating to the sidebar nav.

Specifically, they are "Missing content" errors with details: "Headings and anchors must have an accessible name, which can come from: inner text, aria-label, aria-labelledby, an img with alt property, or an svg with a tag <title></title>."

Error is related to this file:
/node_modules/@astrojs/starlight/components/TableOfContents/TableOfContentsList.astro:17:34

### Link to Minimal Reproducible Example

https://stackblitz.com/github/withastro/starlight/tree/main/examples/basics

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Thanks for the issue @NeilJS. This is not actually an issue in Starlight but an error in Astro’s audit tool. I’ve never dug into exactly why it generates these false reports, but I think it’s something to do with the fact that the links are hidden inside a collapsed menu.
